src/battle_script_commands.c

```diff
@@ -5388,7 +5388,7 @@ static void Cmd_moveend(void)
                     gStatuses3[gBattlerTarget] |= STATUS3_SMACKED_DOWN;
                     gStatuses3[gBattlerTarget] &= ~(STATUS3_MAGNET_RISE | STATUS3_TELEKINESIS | STATUS3_ON_AIR);
                     effect = TRUE;
-                    BattleScriptPush(gBattlescriptCurrInstr + 1);
```

Comment:
This isn't really a fix. We push `gBattlescriptCurrInstr + 1` generally because we want to `return` to the battle script cmd directly after `seteffectwithchance`/`seteffectsecondary` otherwise we end up calling that cmd again. So this still "works" because nothing happens when we're already smacked down, but I think the +1 should stay for consistency with other move effects

Reply:
The original code can't possibly be correct, +1 isn't the next instruction, it's the middle of this one and causes us to interpret the parameters as if they were instructions. +1 is [here](https://github.com/rh-hideout/pokeemerald-expansion/blob/upcoming/asm/macros/battle_script.inc#L416). Did you try the [example in the PR this is copied from](https://github.com/rh-hideout/pokeemerald-expansion/pull/2529#discussion_r1054239892)?
> You can test that it doesn't work correctly by having a Pokémon holding a Life Orb use Smack Down against a non-grounded target; the Life Orb damage is skipped.

Reply:
OOPS I thought this was in `setmoveeffect` not `moveend`. Yeah I can see why the confusion was made originally. This is correct then, sorry